demo/App.tsx

```diff
@@ -50,14 +50,18 @@ class App extends Component<any, any> {
                     }} />
             </div>);
         } else if (mode === AppMode.TaleOfTwoTables) {
-            const props: any = {};
-            Object.entries(this.state.tableProps).forEach(([key, value]) => {
-                props[key] = this.propCache.get(key)(value);
-            });
+            if (!this.state.tableProps2) {
+                this.setState({
+                    tableProps2: R.clone(this.state.tableProps)
+                });
+            }
 
-            return (<DataTable
-                {...props}
-            />);
+            const baseId = this.state.tableProps2 && this.state.tableProps2.id;
+
+            return (this.state.tableProps2 ? <DataTable
+                {...this.state.tableProps2}
+                id={baseId ? 'table2' : baseId}
+            /> : null);
```

Comment:
Change the 2 tables test behavior so that
- the props are not shared between the tables 🙄 
- the 2nd table will have an `id` only if the 1st one has one